tools/optaplanner-rewrite/src/test/java/org/optaplanner/rewrite/v8/AsConstraintBuilderTest.java

```diff
@@ -0,0 +1,2140 @@
+package org.optaplanner.rewrite.v8;
+
+import static org.openrewrite.java.Assertions.java;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+class AsConstraintBuilderTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new AsConstraintBuilder())
+                .parser(JavaParser.fromJavaVersion()
+                        .classpath("optaplanner-core"));
+    }
+
+    // ************************************************************************
+    // Uni
+    // ************************************************************************
+
+    @Test
+    void uniPenalizeName() {
```

Comment:
This is one of the 4 tests that already succeed. In Maven.

Not in IntelliJ... because openrewrite expects the optaplanner-core.jar to be there (which isn't the case in intellij runs). PITA. Looking into that problem too.